src/factories/TextureFactory.cpp

```diff
@@ -4,18 +4,22 @@
 #include "Companion.h"
 #include <iomanip>
 
-static const std::unordered_map <std::string, TextureType> gTextureTypes = {
-    { "RGBA16", TextureType::RGBA16bpp },
-    { "RGBA32", TextureType::RGBA32bpp },
-    { "CI4", TextureType::Palette4bpp },
-    { "CI8", TextureType::Palette8bpp },
-    { "I4", TextureType::Grayscale4bpp },
-    { "I8", TextureType::Grayscale8bpp },
-    { "IA1", TextureType::GrayscaleAlpha1bpp },
-	{ "IA4", TextureType::GrayscaleAlpha4bpp },
-	{ "IA8", TextureType::GrayscaleAlpha8bpp },
-	{ "IA16", TextureType::GrayscaleAlpha16bpp },
-    { "TLUT", TextureType::TLUT },
+extern "C" {
+#include "n64graphics/n64graphics.h"
+}
+
+static const std::unordered_map <std::string, TextureFormat> gTextureTypes = {
```

Comment:
```suggestion
static const std::unordered_map <std::string, TextureFormat> gTextureFormats = {
```

Should update names too otherwise you'll end up with gTextureTypes[var].type which is confusing